libraries/WiFi/src/WiFiSTA.cpp

```diff
@@ -498,3 +498,62 @@ IPv6Address WiFiSTAClass::localIPv6()
     }
     return IPv6Address(addr.addr);
 }
+
+
+bool WiFiSTAClass::_smartConfigStarted = false;
+bool WiFiSTAClass::_smartConfigDone = false;
+
+
+bool WiFiSTAClass::beginSmartConfig() {
+    if (_smartConfigStarted) {
+        return false;
+    }
+
+    if (!WiFi.mode(WIFI_STA)) {
+        return false;
+    }
+
+
+    esp_err_t err;
+    err = esp_smartconfig_start(reinterpret_cast<sc_callback_t>(&WiFiSTAClass::_smartConfigCallback), 1);
+    if (err == ESP_OK) {
+        _smartConfigStarted = true;
+        _smartConfigDone = false;
+        return true;
+    }
+    return false;
+}
+
+bool WiFiSTAClass::stopSmartConfig() {
+    if (!_smartConfigStarted) {
+        return true;
+    }
+
+    if (esp_smartconfig_stop() == ESP_OK) {
+        _smartConfigStarted = false;
+        return true;
+    }
+
+    return false;
+}
+
+bool WiFiSTAClass::smartConfigDone() {
+    if (!_smartConfigStarted) {
+        return false;
+    }
+
+    return _smartConfigDone;

```

Comment:
i do not see _smartConfigDone being set to true anywhere? Maybe missing from the callback?